File: javascript/String/algoLadderStarter/alternateCapitals/alternateCapitals_explained.js
// Given a string, return a copy of the original string that has every other character capitalized. (Capitalization should begin with the second character.)

// Input: “hello, how are your porcupines today?”
// Output: “hElLo, HoW ArE YoUr pOrCuPiNeS ToDaY?”

// Create function that takes in a string
const alternateCapitals = function (string) {
  // Create index that we will use to loop through the array
  let index = 0
  // Create result string
  let result = ""
  // While index is less than the length of the input string
  while (index < string.length) {
    // If the index number is divisible by 2 with no remainder (checking for every other letter here since they will be even) than...
    if (index % 2 === 0) {
      // Add that letter to the results string upper cased
      result = result + string[index].toUpperCase()
      // Else just add that character to the string
    } else {
      result = result + string[index]
    }
    // Increment the index to continue through the array
    index += 1
  }
  // Return the result array
  return result
}
// Call the AlternateCapitals array with the givin input
console.log(alternateCapitals("hello, how are your porcupines today?"))

File: ruby/algo_ladder/alternate_capitals/alternate_capitals_explained.rb
# Given a string, return a copy of the original string that has every other character capitalized. (Capitalization should begin with the second character.)

# Input: “hello, how are your porcupines today?”
# Output: “hElLo, HoW ArE YoUr pOrCuPiNeS ToDaY?”

# Create function that takes in a string
def alternate_capitals (input_string)
  # Create empty results string
  result_string = ""
  # Create index starting at zero
  index = 0
  # Create a while loop that will run the length of the input_string
  while index < input_string.length
    # If the index is odd then add it to the results string upcased
    if index % 2 == 1
      result_string << input_string[index].upcase
      # else add it without upcasing it
    else
      result_string << input_string[index]
    end
    # Increment the index
    index += 1
  end
  # Return the result_string
  return result_string
end
# Call the function with test input
p alternate_capitals("hello, how are your porcupines today?")